client/me/purchases/downgrade/index.tsx

```diff
@@ -129,6 +130,27 @@ export const Downgrade: React.FC< DowngradeProps > = ( props ) => {
 	const isAtomicSiteDowngrade =
 		isAtomicSite && ! targetPlan?.getIncludedFeatures?.().includes( WPCOM_FEATURES_ATOMIC );
 
+	const refundAmount = useMemo( () => {
+		if ( ! purchase ) {
+			return '0';
+		}
+
+		const { refundOptions, currencyCode } = purchase;
+		const defaultFormatter = new Intl.NumberFormat( 'en-US', {
+			style: 'currency',
+			currency: currencyCode,
+		} );
+		const precision = defaultFormatter.resolvedOptions().maximumFractionDigits;
+		const amount =
+			isRefundable( purchase ) &&
+			Array.isArray( refundOptions ) &&
+			refundOptions[ 0 ]?.refund_amount
```

Comment:
I debugged a bit and I think this is the reason it's showing the wrong refund amount (see my earlier comment).  The code is using the first element of the array here, but it actually needs to pick the correct element of the array depending on which product is being downgraded to.

As you mentioned, you basically just copied this code from the existing self-serve downgrade flow within cancellation, so a version of the bug exists there too.  The code has been there for years, but perhaps it never mattered much in practice (until recently when the server started returning more downgrade options than it did before).  Eventually it will need to be fixed in both places.

Reply:
Thanks for debugging this. I added a search to match `to_product_id` to the target plan's.